### PR TITLE
VP-3927 Refactor search builder

### PR DESCRIPTION
**Pull request by Vectorfield4**
https://virtocommerce.atlassian.net/browse/VP-3927
https://virtocommerce.atlassian.net/browse/VP-3611

### Proposed of changes
Describe the technical details of realization provided by you.

### Make sure these boxes are checked:
- [x] Check all the changes in github PR - files count (non of them are redundant, have meaningful changes, all are added), if target branch is correct
- [x] Check methods and variable namings - it should be self descriptive, no typos
- [x] Check you did not introduce breaking changes in API and public models/services.
- [x] Respect extensibility - https://community.virtocommerce.com/t/extensibility-basics-the-domain-model-and-persistence-layer-extension/141
- [x] Follow [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and [SOLID](https://en.wikipedia.org/wiki/SOLID) principles
- [x] For unit tests - follow Microsoft best practices: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices
- [x] Consolidate solution dependencies in case you are using newer version, update VC module dependencies in module.manifest. Do not upgrade 3rd party packages that are shipped with the platform with the version newer than in the platform.
- [x] Check code style conventions - https://github.com/VirtoCommerce/styleguide/blob/master/csharp.md
- [x] Check PR have a concise and descriptive title, follow [git commit message rules](https://github.com/VirtoCommerce/styleguide/blob/master/gitcommits.md)

### CODE REVIEW

File: src/VirtoCommerce.ExperienceApiModule.Core/Extensions/MetaDataProviderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                aliasContainer = new AliasContainer();
+                provider.Metadata.Add("aliases", aliasContainer);
+            }
+
+            aliasContainer.AddRootAliases(values);
+
+            return provider;
+        }
+
+        public static IProvideMetadata InnerAliase(this IProvideMetadata provider, string value)
```
Comment (yecli) [inline src/VirtoCommerce.ExperienceApiModule.Core/Extensions/MetaDataProviderExtensions.cs:21]: InnerAliase - typo?

File: src/VirtoCommerce.ExperienceApiModule.Core/Index/ElasticSearchRequestBuilder.cs [hunk 2]
```diff
@@ -3,29 +3,44 @@
...
 using VirtoCommerce.CatalogModule.Core.Search;
-using VirtoCommerce.ExperienceApiModule.DigitalCatalog.Index;
 using VirtoCommerce.Platform.Core.Common;
 using VirtoCommerce.SearchModule.Core.Model;
 using VirtoCommerce.SearchModule.Core.Services;
 
 namespace VirtoCommerce.ExperienceApiModule.Core.Index
 {
-    public class SearchRequestBuilder
+    public interface IRequestBuilder
```
Comment (yecli) [inline src/VirtoCommerce.ExperienceApiModule.Core/Index/ElasticSearchRequestBuilder.cs:12]: Could it be moved to the separate file?

File: src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/Aliases/AliasContainer.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        public void AddRootAliases(params string[] values)
+        {
+            foreach (var value in values.Where(value => !_rootAliases.Any(y => y.Value.EqualsInvariant(value))))
+            {
+                _rootAliases.Add(new RootAlias(value));
+            }
+        }
+
+        public void SetInnerAliase(string value)
```
Comment (yecli) [inline src/VirtoCommerce.ExperienceApiModule.Core/Infrastructure/Aliases/AliasContainer.cs:27]: SetInnerAliase - typo?

File: src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Queries/LoadCategoryQuery.cs [hunk 4]
```diff
@@ -1,13 +1,13 @@
 using System;
 using System.Collections.Generic;
+using VirtoCommerce.ExperienceApiModule.Core.Index;
 using VirtoCommerce.ExperienceApiModule.Core.Infrastructure;
-using VirtoCommerce.XDigitalCatalog.Interfaces;
 
 namespace VirtoCommerce.XDigitalCatalog.Queries
 {
-    public class LoadCategoryQuery : IQuery<LoadCategoryResponce>, IHasIncludeFields
+    public class LoadCategoryQuery : IQuery<LoadCategoryResponce>, IHasIncludeFields, IGetSingleDocumentQuery
```
Comment (yecli) [inline src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Queries/LoadCategoryQuery.cs:8]: LoadCategoryRespon**c**e - typo

File: src/VirtoCommerce.ExperienceApiModule.Core/Index/IGetSingleDocumentQuery.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿namespace VirtoCommerce.ExperienceApiModule.Core.Index
+{
+    public interface IGetSingleDocumentQuery : ISearchQuery, IHaveObjectId, IHasIncludeFields
```
Comment (yecli) [inline src/VirtoCommerce.ExperienceApiModule.Core/Index/IGetSingleDocumentQuery.cs:3]: Do we need this interface in addition to `IGetAllDocumentsByIdsQuery`?
Isn't it just a specific case?

**Comment by Vectorfield4** [comment]
Merge this with https://github.com/VirtoCommerce/vc-storefront/pull/498/

File: src/VirtoCommerce.ExperienceApiModule.Core/Index/IHaveObjectId.cs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+﻿namespace VirtoCommerce.ExperienceApiModule.Core.Index
```
Comment (yecli) [inline src/VirtoCommerce.ExperienceApiModule.Core/Index/IHaveObjectId.cs:1]: This is not needed anymore

**Pull request merged**